packages/graphql/src/schema/augment/fulltext.ts

```diff
@@ -17,50 +17,114 @@
 
+function createEnumType(name: string, rawValues: string[]): GraphQLEnumType {
+    const values = rawValues.reduce((acc, value) => {
+        return {
+            ...acc,
+            [value]: {
+                value,
+            },
+        };
+    }, {} as GraphQLEnumValueConfigMap);
+
+    return new GraphQLEnumType({
+        name: `${name}FulltextIndexes`,
+        values,
+    });
+}
+
 export function augmentFulltextSchema(
     node: Node,
     composer: SchemaComposer,
     nodeWhereTypeName: string,
     nodeSortTypeName: string
 ) {
     if (node.fulltextDirective) {
-        const fields = node.fulltextDirective.indexes.reduce((res, index) => {
+        const indexes = node.fulltextDirective.indexes.map((index) => {
+            const indexName = index.indexName || index.name;
+            if (indexName === undefined) {
+                throw new Error("The name of the fulltext index should be defined using the indexName argument.");
+            }
+            return indexName;
+        });
+
+        const indexEnumType = createEnumType(node.name, indexes);
+        const indexEnum = composer.createEnumTC(indexEnumType);
+
+        const legacyFields = node.fulltextDirective.indexes.reduce((res, index) => {
             const indexName = index.indexName || index.name;
             if (indexName === undefined) {
                 throw new Error("The name of the fulltext index should be defined using the indexName argument.");
             }
+
+            composer.createInputTC({
+                name: `${node.name}${upperFirst(indexName)}Fulltext`,
+                fields: {
+                    phrase: new GraphQLNonNull(GraphQLString),
+                },
+            });
+
             return {
                 ...res,
-                [indexName]: composer.createInputTC({
-                    name: `${node.name}${upperFirst(indexName)}Fulltext`,
-                    fields: {
-                        phrase: new GraphQLNonNull(GraphQLString),
-                    },
-                }),
+                [indexName]: {
+                    type: `${node.name}${upperFirst(indexName)}Fulltext`,
+                    directives: [
+                        {
+                            name: "deprecated",
+                            args: {
+                                reason: fulltextArgDeprecationMessageInArgument,
+                            },
+                        },
+                    ],
+                },
             };
         }, {});
 
         const fulltextResultDescription = `The result of a fulltext search on an index of ${node.name}`;
         const fulltextWhereDescription = `The input for filtering a fulltext query on an index of ${node.name}`;
         const fulltextSortDescription = `The input for sorting a fulltext query on an index of ${node.name}`;
 
+        // Sets a default index if there is only one index available
+        let defaultIndex = undefined as any;
```

Comment:
```suggestion
        let defaultIndex: string | undefined = undefined;
```

Let's make sure we don't add `any` if possible